### PR TITLE
Add libgc_internal crate to core library

### PR DESCRIPTION
**Pull request by jacob-hughes**

### CODE REVIEW

**Comment by jacob-hughes** [comment]
Rustc is quite particular about `Cargo.lock`, try this: 9850ee61666ebd2b07a140b1bee5d40eceb3b688

**Comment by ltratt** [comment]
Please squash.

**Comment by jacob-hughes** [comment]
Squashed

**Comment by jacob-hughes** [comment]
I've pushed a fix for this, but I think it works better as a separate commit so I won't squash it.

**Comment by jacob-hughes** [comment]
The license checker doesn't look for license files in the repo. I think it just checks for a `license = ` line in `Cargo.toml`. I didn't know about this so I've added a PR [here](https://github.com/softdevteam/libgc_internal/pull/3).

**Comment by jacob-hughes** [comment]
Fixed this, it will need squashing

bors try

**Comment by jacob-hughes** [comment]
It builds! Let me know when I can squash

File: src/tools/tidy/src/extdeps.rs [hunk 1]
```diff
@@ -25,6 +25,12 @@ pub fn check(root: &Path, bad: &mut bool) {
         // Extract source value.
         let source = line.split_once('=').unwrap().1.trim();
 
+        // rustc only permits crates from crates.io so for now libgc_internal is
+        // hardcoded to pass the tidy checker.
+        if source.contains("softdevteam/libgc_internal") {
```
Comment (ltratt) [inline src/tools/tidy/src/extdeps.rs:30]: Can we make this `ends_with` or similar, just so that it's less likely to trigger incorrectly?

Comment (jacob-hughes) [reply src/tools/tidy/src/extdeps.rs:30]: Unfortunately not. The actual sources list looks something like this:`<metadata>softdevteam/libgc_internal<hash_which_changes_when_package_updates>`

Comment (ltratt) [reply src/tools/tidy/src/extdeps.rs:30]: Ah, OK.

**Comment by ltratt** [comment]
Please squash.

**Comment by jacob-hughes** [comment]
Squashed

**Pull request merged**